benchmarks/cugraph/pytest-based/bench_algos.py

```diff
@@ -268,6 +268,25 @@ def is_graph_distributed(graph):
     return isinstance(graph.edgelist.edgelist_df, dask_cudf.DataFrame)
 
 
+def get_vertex_pairs(G, subset_factor=3):
+    """
+    Return a DF of two columns representing random vertex pairs
+    """
+    sample_num = G.number_of_nodes() // subset_factor
+    col_a = (
+        G.select_random_vertices(num_vertices=sample_num)
```

Comment:
You can instead call [get_two_hop_neighbors](https://github.com/rapidsai/cugraph/blob/branch-24.08/python/cugraph/cugraph/structure/graph_implementation/simpleDistributedGraph.py#L853) which returns `vertex_pairs` based on a given set of `vertices`. You can still randomly generate the set of vertices with `select_random_vertices` and pass it to `get_two_hop_neighbors`

Reply:
Ah, thank you for the tip. I'll check that out.

Reply:
@jnke2016 I understand that `get_two_hop_neighbors` takes in a subset of `node_ids` from the Graph and returns a sorted list of all two-hop neighbor pairs. I was wondering what the advantage/goal of using this instead of constructing a random `src`/`dst` pair like I am doing now? Is this something to do with how the algs work?

Reply:
>  Is this something to do with how the algs work?

Not necessarily but we already have and efficient code generating pairs. I don't see the use of creating a new one.